core/src/main/java/org/apache/accumulo/core/file/rfile/GenerateSplits.java

```diff
@@ -201,7 +201,9 @@ public void execute(String[] args) throws Exception {
       itemsSketch.update(row);
       iterator.next();
     }
-    return itemsSketch.getQuantiles(numSplits);
+    Text[] items = itemsSketch.getQuantiles(numSplits + 2);
```

Comment:
ItemSketch has a method named getRetainedItems(), I wonder if this is the max that could be requested for getQuantiles()?  If so, what should be done if numSplits+2>itemsSketch.getRetainedItems()

Reply:
At that point all we can do is throw an error. I think checking the different outputs of the Datasketches API is a bit out of scope for this utility.

Reply:
> I think checking the different outputs of the Datasketches API is a bit out of scope for this utility.

I was not proposing checking different outputs of the API at this point.  Was more thinking it would be good to fully understand the functionality behind the API and what the edge cases may be. The `.getRetainedItems()` API seems to indicate that that there may be an upper bound to how many splits the API would track, but I am not sure about this.  The class seems to buffer information in memory, but I don't have a good sense of what if any limits there are on how much it will buffer.

Reply:
> The class seems to buffer information in memory, but I don't have a good sense of what if any limits there are on how much it will buffer.

That is a good question, I am not sure either. Perhaps this utility is a good, harmless way to test the limits of the library, in the case where we want to use it in other, more important parts of the code.